### PR TITLE
Fix duplicated inlined images; fixes #5844

### LINKED ISSUE #5844: Corporate email footer showing several repeated images in ticket description
When a user creates a ticket via email the ticket description shows several footer repeated images in ticket description.

github_glpi

Installation:

Información acerca de la instalación y configuración del sistema
--
[code]   GLPI 9.4.0 ( => /var/www/html) Installation mode: TARBALL

Operating system: Linux phdesk01v 3.10.0-957.5.1.el7.x86_64 #1 SMP Fri Feb 1 14:54:57 UTC 2019 x86_64 PHP 7.3.2 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bcmath, bz2, 	calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, json, ldap, libxml, mbstring, 	mcrypt, mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, posix, session, shmop, sockets, sqlite3, standard, sysvmsg, 	sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib) Setup: max_execution_time="180" memory_limit="1024M" post_max_size="10M" safe_mode="" session.save_handler="files" 	upload_max_filesize="10M"  Software: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/7.3.2 () 	Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko Server Software: MySQL Community Server (GPL) 	Server Version: 5.7.25 	Server SQL Mode:  	Parameters: admin@localhost/helpdesk 	Host info: Localhost via UNIX socket 	 mysqli extension is installed ctype extension is installed fileinfo extension is installed json extension is installed mbstring extension is installed iconv extension is installed zlib extension is installed curl extension is installed gd extension is installed simplexml extension is installed xml extension is installed ldap extension is installed imap extension is installed Zend OPcache extension is installed APCu extension is installed xmlrpc extension is installed CAS extension is installed Database version seems correct (5.7.25) - Perfect! /var/www/html/files/_log : OK /var/www/html/config : OK /var/www/html/files : OK /var/www/html/files/_dumps : OK /var/www/html/files/_sessions : OK /var/www/html/files/_cron : OK /var/www/html/files/_graphs : OK /var/www/html/files/_lock : OK /var/www/html/files/_plugins : OK /var/www/html/files/_tmp : OK /var/www/html/files/_cache : OK /var/www/html/files/_rss : OK /var/www/html/files/_uploads : OK /var/www/html/files/_pictures : OK Web access to the files directory should not be allowed but this cannot be checked automatically on this instance. Make sure acces to error log file is forbidden; otherwise review .htaccess file and web server configuration.SELinux mode is Disabled

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #5844 

If multiple entries exists in documents for a same inlined image, inlined tag should be replaced by only one document entry.

### CODE REVIEW

**Comment by trasher** [comment]
Could you confirm this still work with several images expected?

Adding tests on this method would also help.

**Comment by cedric-anne** [comment]
> Could you confirm this still work with several images expected?
> 
Yes, it works as expected.

> Adding tests on this method would also help.
I will look at this.

**Comment by cedric-anne** [comment]
I switch to another solution.

When searching how it could be possible to have multiple documents related to the same tag, I found that GLPI recreates a new document if entity differs (seems logical to prevent visibility issues). So, to be sure that the produced link refers to the good document, I created a similar method that takes the document id as a param, instead of the tag.

File: inc/html.class.php [hunk 1]
```diff
@@ -5695,15 +5695,16 @@ static function jsAlertCallback($msg, $title, $okCallback = null) {
...
     * @param int          $width     witdh of the final image
     * @param int          $height    height of the final image
     * @param bool         $addLink   boolean, do we need to add an anchor link
     * @param string       $more_link append to the link (ex &test=true)
     *
-    * @return nothing
+    * @return string
+    *
+    * @since 9.2
+    * @deprecated 9.4.3
```
Comment (trasher) [inline inc/html.class.php]: You should add a `Toolbox::deprecated()`

**Pull request merged**